modules/videopress/videopress.php

```diff
@@ -731,6 +732,32 @@ function videopress_shortcode_options( $options ) {
 		return $options;
 	}
 
+	/**
+	 * An override for the attachment url, which returns back the WPCOM videopress original url,
+	 * if it is set to the the objects metadata. this allows us to show the original uploaded
+	 * file on the WPCOM architecture, instead of the locally uplodaded file,
+	 * which doeasn't exist.
+	 *
+	 * @param string $url
+	 * @param int $post_id
+	 *
+	 * @return mixed
+	 */
+	public function update_attachment_url_for_videopress( $url, $post_id ) {
+
+		$post = get_post( $post_id );
+
+		if ( $post->post_mime_type === 'video/videopress' ) {
+			$meta = wp_get_attachment_metadata( $post->ID );
+
+			if ( isset( $meta['original']['url'] ) ) {
```

Comment:
It technically needs to do more than that.

It also needs to send the file up to the API to have it transcoded. However, we already have a media item in this case, so it doesn’t fit exactly. However, once that transcode runs, it will automatically update the media item.

My current thinking is to have a variation to the upload endpoint that this will use that does most of the same work, but attaches it to an existing id, instead of creating a new one. After that, everything else should work itself out.

Drew Butler

On Mon, Jun 20, 2016 at 4:31 PM George Stephanis

<
mailto:George Stephanis <notifications@github.com>
> wrote:

In
https://github.com/Automattic/jetpack/pull/4146#discussion_r67770510
:

> + * file on the WPCOM architecture, instead of the locally uplodaded file, > + * which doeasn't exist. > + * > + * @param string $url > + * @param int $post_id > + * > + * @return mixed > + */ > + public function update_attachment_url_for_videopress( $url, $post_id ) { > + > + $post = get_post( $post_id ); > + > + if ( $post->post_mime_type === 'video/videopress' ) { > + $meta = wp_get_attachment_metadata( $post->ID ); > + > + if ( isset( $meta['original']['url'] ) ) {

What's your thinking on the

create_local_media_library_for_videopress_guid()

function -- should we update it so it just calls

xmlrpc_update_videopress_info()

instead of doing the work itself?
https://github.com/Automattic/jetpack/blob/master/modules/videopress/utility-functions.php#L142

—

You are receiving this because you authored the thread.

Reply to this email directly,
https://github.com/Automattic/jetpack/pull/4146/files/c4b505af4f238979ce6e9d6ea714b97134c0cdab#r67770510
, or
https://github.com/notifications/unsubscribe/AAL6F5N6ZuSgobTlAmnoOuGrRVJayXI-ks5qNwaygaJpZM4I3qFK
.

https://github.com/Automattic/jetpack/pull/4146/files/c4b505af4f238979ce6e9d6ea714b97134c0cdab#r67770510